src/core/src/service_interfaces/LifecycleManagerAzure.py

```diff
@@ -0,0 +1,124 @@
+            if int(self.execution_config.sequence_number) > int(extension_sequence['number']) \
+                    or int(self.execution_config.sequence_number) > int(core_sequence['number']):
+                self.composite_logger.log_error("Auto-assessment not started as an extension state anomaly was detected. [Attempted={0}][DetectedExt={1}][DetectedCore={2}]".format(str(self.execution_config.sequence_number), str(extension_sequence['number']),str(core_sequence['number'])))
+                self.env_layer.exit(0)
+
+            # attempted sequence number is same as recorded core sequence - expected
+            if int(self.execution_config.sequence_number) == int(core_sequence['number']):
+                if bool(core_sequence['completed']):
+                    self.composite_logger.log_debug("Auto-assessment is safe to start. Existing sequence number marked as completed.")
+                    self.update_core_sequence(completed=False)  # signalling core restart with auto-assessment as its safe to do so
+                else:
+                    self.composite_logger.log_debug("Auto-assessment may not be safe to start yet as core sequence is not marked completed.")
+                    if len(self.identify_running_processes(core_sequence['process_ids'])) != 0:
+                        # NOT SAFE TO START
+                        # Possible reasons: full core operation is in progress (okay), some previous auto-assessment is still running (bad scheduling, adhoc run, or process stalled)
+                        self.composite_logger.log_warning("Auto-assessment is NOT safe to start yet. Existing core process(es) running. Exiting. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
+                        self.env_layer.exit(0)
+                    else:
+                        # MAY BE SAFE TO START
+                        self.composite_logger.log_warning("Auto-assessment is LIKELY safe to start, BUT core sequence anomalies were detected. Evaluating further. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
+
+                        # wait to see if Core comes back from a restart
+                        timer_start_time = self.env_layer.datetime.datetime_utcnow()
+                        while True:
+                            core_sequence = self.read_core_sequence()
+
+                            # Main Core process suddenly started running (expected after reboot) - don't run
+                            if len(self.identify_running_processes(core_sequence['process_ids'])) != 0:
+                                self.composite_logger.log_warning("Auto-assessment is NOT safe to start as core process(es) started running. Exiting. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
+                                self.env_layer.exit(0)
+
```

Comment:
There may be opportunities to refactor common code into separate functions in the parent class. Not for this PR.